### PR TITLE
Corrected Vellum and Vanish item bonuses

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: #4016

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Cleaned up the way Vellum damage and Vanish damage are calculated.
  * Implemented bonus3 bSPVanishRate/bHPVanishRate.
  * Updated Mental Destroyer item bonus.
  * Cleaned up and added documentation.
Thanks to @Balferian!

### CODE REVIEW

File: doc/item_bonus.txt [hunk 1]
```diff
@@ -375,10 +375,12 @@ bonus2 bSPDrainRate,x,n;		Adds a x/10% chance to drain n% SP from inflicted dama
 HP/SP vanish
 ------------
 bonus2 bHPVanishRate,x,n;		Add a x/10% chance of decreasing enemy's HP amount by n% when attacking
-bonus2 bSPVanishRate,x,n;		Add a x/10% chance of decreasing enemy's SP amount by n% when attacking
-
 bonus3 bHPVanishRaceRate,r,x,n;		Add a x/100% chance of decreasing enemy's HP amount by n% when attacking, depends on enemy race r
+bonus3 bHPVanishRate,x,n,bf;	Add a x/10% chance of decreasing enemy's HP amount by n% when attacking with trigger criteria bf
+
+bonus2 bSPVanishRate,x,n;		Add a x/10% chance of decreasing enemy's SP amount by n% when attacking
 bonus3 bSPVanishRaceRate,r,x,n;		Add a x/100% chance of decreasing enemy's SP amount by n% when attacking, depends on enemy race r
```
Comment (Atemo) [inline doc/item_bonus.txt]: `x/10%` for `bHPVanishRaceRate` and `bSPVanishRaceRate`

File: src/map/battle.cpp [hunk 2]
```diff
@@ -7006,40 +7010,30 @@ bool battle_vanish(struct map_session_data *sd, struct block_list *target, struc
...
 
-	memset(&d, 0, sizeof(d));
-
-	// Check for vanish HP/SP. !CHECKME: Which first, drain or vanish?
-	battle_vanish(sd, tbl, &d);
-
-	// Check for drain HP/SP
-	hp = sp = i = 0;
-	for (i = 0; i < 4; i++) {
+	for (int i = 0; i < 4; i++) {
```
Comment (Normynator) [inline src/map/battle.cpp]: Can't we use symbolic constants for better readability and an iterator?

(exmaple here: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es45-avoid-magic-constants-use-symbolic-constants)

File: src/map/battle.cpp [hunk 3]
```diff
@@ -7297,19 +7291,18 @@ enum damage_lv battle_weapon_attack(struct block_list* src, struct block_list* t
...
 	if( sc && sc->count ) {
 		if (sc->data[SC_EXEEDBREAK]) {
-			if (!is_infinite_defense(target, wd.flag) && !vanish_damage)
+			if (!is_infinite_defense(target, wd.flag) && !vellum_damage)
 				wd.damage *= sc->data[SC_EXEEDBREAK]->val2 / 100;
 			status_change_end(src, SC_EXEEDBREAK, INVALID_TIMER);
 		}
 		if( sc->data[SC_SPELLFIST] ) {
-			if( --(sc->data[SC_SPELLFIST]->val1) >= 0 && !vanish_damage ){
+			if( --(sc->data[SC_SPELLFIST]->val1) >= 0 && !vellum_damage ){
```
Comment (Normynator) [inline src/map/battle.cpp]: Wouldn't
` if( sc->data[SC_SPELLFIST]->val1 > 0 && !vellum_damage ){`
be much easier to read?

Comment (aleos89) [reply src/map/battle.cpp]: We have to decrement the value of `val1` which is the number of physical hits remaining.

File: src/map/battle.cpp [hunk 4]
```diff
@@ -6828,6 +6853,12 @@ struct Damage battle_calc_attack(int attack_type,struct block_list *bl,struct bl
 	}
 	else // Some skills like Weaponry Research will cause damage even if attack is dodged
 		d.dmg_lv = ATK_DEF;
+
+	struct map_session_data *sd = BL_CAST(BL_PC, bl);
+
+	if (sd && d.damage + d.damage2 > 1)
```
Comment (Demintika) [inline src/map/battle.cpp]: You changed the condition to trigger Vanish damage from "greater than 0 damage" to "greater than 1 damage". Is there any bug if it's the former? I need it in my server.

**Pull request merged**